tests/integration/test_remote_write_grafana_agent.py

```diff
@@ -47,6 +48,60 @@ async def test_remote_write_with_grafana_agent(ops_test, prometheus_charm):
     )
 
 
+@pytest.mark.abort_on_fail
+async def test_check_data_persist_on_kubectl_delete_pod(ops_test, prometheus_charm):
+    prometheus_app_name = "prometheus"
+    pod_name = f"{prometheus_app_name}-0"
+    query = "prometheus_tsdb_head_chunks_created_total{}"
+    total0 = await run_promql(ops_test, query, prometheus_app_name)
+    sum0 = int(total0[0]["value"][1])
+
+    cmd = [
+        "sg",
+        "microk8s",
+        "-c",
+        " ".join(["microk8s.kubectl", "delete", "pod", "-n", ops_test.model_name, pod_name]),
+    ]
+
+    logger.debug(
+        "Removing pod '%s' from model '%s' with cmd: %s", pod_name, ops_test.model_name, cmd
+    )
+
+    retcode, stdout, stderr = await ops_test.run(*cmd)
+    assert retcode == 0, f"kubectl failed: {(stderr or stdout).strip()}"
+    logger.debug(stdout)
+    await ops_test.model.block_until(
+        lambda: len(ops_test.model.applications[prometheus_app_name].units) > 0
+    )
+    await ops_test.model.wait_for_idle(apps=[prometheus_app_name], status="active", timeout=60)
+    assert check_prometheus_is_ready(ops_test, prometheus_app_name, 0)
+
+    total1 = await run_promql(ops_test, query, prometheus_app_name)
+    sum1 = int(total1[0]["value"][1])
+    assert sum0 <= sum1
+
+
+@pytest.mark.abort_on_fail
+async def test_check_data_persist_on_charm_upgrade(ops_test, prometheus_charm):
```

Comment:
Should this be moved into `test_upgrade_charm.py`?

Reply:
That make sense!